pkg/controller/deployment/deployment_controller.go

```diff
@@ -80,33 +81,18 @@ func (r *ReconcileDeployment) Reconcile(request reconcile.Request) (reconcile.Re
 		// Error reading the object - requeue the request.
 		return reconcile.Result{}, err
 	}
+	pods := &v1.JaegerList{}
+	err = r.client.List(context.Background(), pods)
+	if err != nil {
+		log.WithError(err).Error("failed to get the available Jaeger pods")
+		return reconcile.Result{}, err
+	}
 
-	if inject.Needed(instance) {
-		pods := &v1.JaegerList{}
-		opts := []client.ListOption{}
-		err := r.client.List(context.Background(), pods, opts...)
-		if err != nil {
-			log.WithError(err).Error("failed to get the available Jaeger pods")
-			return reconcile.Result{}, err
-		}
+	injected := inject.IfNeeded(instance, pods)
 
-		jaeger := inject.Select(instance, pods)
-		if jaeger != nil && jaeger.GetDeletionTimestamp() == nil {
-			// a suitable jaeger instance was found! let's inject a sidecar pointing to it then
-			// Verified that jaeger instance was found and is not marked for deletion.
-			log.WithFields(log.Fields{
-				"deployment":       instance.Name,
-				"namespace":        instance.Namespace,
-				"jaeger":           jaeger.Name,
-				"jaeger-namespace": jaeger.Namespace,
-			}).Info("Injecting Jaeger Agent sidecar")
-			instance = inject.Sidecar(jaeger, instance)
-			if err := r.client.Update(context.Background(), instance); err != nil {
-				log.WithField("deployment", instance).WithError(err).Error("failed to update")
-				return reconcile.Result{}, err
-			}
-		} else {
-			log.WithField("deployment", instance.Name).Info("No suitable Jaeger instances found to inject a sidecar")
+	if injected != nil {
+		if err := r.client.Update(context.Background(), injected); err != nil {
```

Comment:
It would probably be better to create a context at the beginning of the reconcile method, and use the same context everywhere down the stream. It will make it easier once we decide to add tracing to this controller, like we added to the other controller.